src/ucs/arch/riscv/cpu.c

```diff
@@ -0,0 +1,72 @@
+/**
+* Copyright (C) Tactical Computing Labs, LLC. 2022. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#if defined(__riscv)
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucs/arch/cpu.h>
+#include <stdio.h>
+
+static void ucs_rv64_cpuid_from_proc(ucs_rv64_cpuid_t *cpuid)
+{
+    int count = 0;
+    char buf[256];
+    char value[256] = {0};
+    FILE* f;
+
+    f = fopen("/proc/cpuinfo","r");
+    if (!f) {
+        return;
+    }
+
+    while (fgets(buf, sizeof(buf), f)) {
+        if (sscanf(buf, "uarch           : %s", value) == 1) {
+                memcpy(cpuid->uarch, value, 256); //printf("%s\n", value);
+                ++count;
+        }
+        else if (sscanf(buf, "isa           : %s", value) == 1) {
+                memcpy(cpuid->isa, value, 256); //printf("%s\n", value);
+                ++count;
+        } else if (sscanf(buf, "mmu           : %s", value) == 1) {
+                memcpy(cpuid->mmu, value, 256); //printf("%s\n", value);
+                ++count;
+        }
+
+        if (count == 3) {
+            break;
+        }
+    }
+
+    fclose(f);
+}
+
+void ucs_rv64_cpuid(ucs_rv64_cpuid_t *cpuid)
```

Comment:
replace rv64 by riscv in all places

Reply:
@shamisp 's comment about riscv or rv64 applies here. will need to sort out naming convention.